File: tests/fixtures/negative/security/JdbcStatementExecute.java
package test;

import java.sql.Connection;
import java.sql.PreparedStatement;

class JdbcStatementExecuteJavaSafeFixture {
    void load(Connection connection) throws Exception {
        connection.createStatement().executeQuery("SELECT * FROM users");
        PreparedStatement stmt = connection.prepareStatement("SELECT * FROM users WHERE id = ?");
        stmt.executeQuery();
    }
}

File: tests/fixtures/negative/security/JdbcStatementExecute.kt
package test

import java.sql.Connection

private const val USERS_TABLE = "users"
private const val COLUMN_ID = "id"

class JdbcStatementExecuteSafeFixture {
    fun load(connection: Connection) {
        connection.createStatement().executeQuery("SELECT * FROM users")
        connection.createStatement().executeQuery("SELECT * FROM $USERS_TABLE WHERE $COLUMN_ID = 1")
        val prepared = connection.prepareStatement("SELECT * FROM users WHERE id = ?")
        prepared.executeQuery()
    }
}
